File: GeneradorClases_V2/Plantillas/ReglasNegocio.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Data;

namespace <Namespace>
{
    public class <TableName>
    {
        Datos.<TableName> d<TableName>;

        public <TableName>()
        {
            d<TableName> = new Datos.<TableName>();
        }

        public int Crear(Objetos.<TableName> o<TableName>)
        {
            return d<TableName>.Crear(o<TableName>);
        }

        public void Actualizar(Objetos.<TableName> o<TableName>)
        {
            d<TableName>.Actualizar(o<TableName>);
        }

        public Objetos.<TableName> Consultar(int id)
        {
            return ConvertirDesdeDataTable(d<TableName>.Consultar(id)); 
        }

        public DataTable Listar()
        {
            return d<TableName>.Listar();
        }

        public void Retirar(int id)
        {
            d<TableName>.Retirar(id); 
        }
		
		public Objetos.<TableName> ConvertirDesdeDataTable(DataTable dt<TableName>)
        {
            Objetos.<TableName> o<TableName> = null;
            if (dt<TableName>.Rows.Count > 0)
            {
                o<TableName> = new Objetos.<TableName>();
                System.Reflection.PropertyInfo[] Campos = o<TableName>.GetType().GetProperties();
                for (int i = 0; i < Campos.Length; i++)
                {
					try{
						Campos[i].SetValue(o<TableName>, dt<TableName>.Rows[0][Campos[i].Name], null);
					}catch(Exception ex){}
                }
            }
            return o<TableName>;
        }
    }
}

File: GeneradorClases_V2/Plantillas/ReglasNegocio.vb
Imports System
Imports System.Text
Imports System.Data
Imports Comfama.Objetos
Imports Comfama.Datos

Namespace Comfama.ReglasNegocio
    Public Class ReglasNegocio
        Public Function Crear(ByVal oObjetos As Objetos.Objetos) As Integer
            Dim dObjets As New Datos.Datos
            dObjets.Crear(oObjetos)
        End Function

        Public Sub Actualizar(ByVal oObjetos As Objetos.Objetos)
            Dim dObjets As New Datos.Datos
            dObjets.Actualizar(oObjetos)
        End Sub

        Public Function Consultar(ByVal id As Integer) As DataTable
            Dim dObjets As New Datos.Datos
            dObjets.Consultar(id)
        End Function

        Public Function Listar() As DataTable
            Dim dObjets As New Datos.Datos
            dObjets.Listar()
        End Function

        Public Sub Retirar(ByVal id As Integer)
            Dim dObjets As New Datos.Datos
            dObjets.Retirar(id)
        End Sub
    End Class
End Namespace
